src/main/java/com/gargoylesoftware/htmlunit/javascript/host/dom/DOMException.java

```diff
@@ -114,6 +114,9 @@ public class DOMException extends HtmlUnitScriptable {
     /** Data clone error. */
     @JsxConstant
     public static final short DATA_CLONE_ERR = 25;
+    /** Not allowed error. */
+    @JsxConstant
+    public static final short NOT_ALLOWED_ERR = 26;
```

Comment:
marking this as JsxConstant will make this visible to js - at least when running com.gargoylesoftware.htmlunit.general.HostConstantsTest with real browsers you can see that this is not available in the current browsers. Even https://webidl.spec.whatwg.org/#idl-DOMException does not list this.

Reply:
Yeah, I need to write a test to discover how "real" browsers report this error. Does my basic approach in this PR look correct, though.

Reply:
Yes the rest looks like you are on the right path :-)

Reply:
@rbri Is there any documentation on event registration and propagation in HtmlUnit? I'm not making any headway on my revisions because I can't figure out how to add support for the `slot-change` event and have no idea where to start.
I also keep getting confused about delineation between the browser DOM and its JavaScript representation in the structure of this project. Do you have any block diagrams, data flow diagrams, design documents, or other artifacts that could help me navigate the code more smoothly?

Reply:
Seems like the correct way to handle this might be something like this:
```
throw Context.reportRuntimeError("NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.");
```

Reply:
@rbri I've been poking at this PR as often as I can, but progress is slow. I'm trying to figure out how everything fits together, but there seems to be at least three distinct layers to the object hierarchy and trying to keep them straight sometimes makes me dizzy.